lib/frontend.ml

```diff
@@ -163,7 +164,8 @@ module Make(C: S.CONFIGURATION with type 'a io = 'a Lwt.t) = struct
       let pages = Io_page.pages num in
       List.iter
         (fun (gref, page) ->
-           let id = gref mod (1 lsl 16) in
+           let id = nf.rx_id in
+           nf.rx_id <- (succ nf.rx_id) mod (1 lsl 16) ;
```

Comment:
It depends how rigorous you want to be here. Ideally, I'd make something similar to Shared_page_pool, but for receiving (with full pages). That would probably speed things up a bit too, since sharing and unsharing pages is slow. It's more work, of course :-/

Or you could just push the IDs of released pages onto a stack and pop them off when needed instead of minting new ones.